main_test.go

```diff
@@ -0,0 +1,377 @@
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs([]string{"-l", "-p",
+			"testdata/patch/error1.patch", "testdata/test_files/test2.go"})
+		require.NoError(t, err)
+		path := "testdata/patch/error1.patch"
+		_, err = loadPatches(command.fset, opts, os.Stdin)
+		expectedErrorString := "could not read \"" + path + "\": " + "open " + path + ": no such file or directory"
+		require.EqualError(t, err, expectedErrorString)
+	})
+}
+
+func TestApply(t *testing.T) {
+	t.Parallel()
+	t.Run("Verify functionality of Apply when successful", func(t *testing.T) {
+		argParser, opts := newArgParser()
+		_, err := argParser.ParseArgs(command.args)
+		filename := "testdata/test_files/lint_example/test2.go"
+		content, err := ioutil.ReadFile(filename)
+		fast, err := parser.ParseFile(command.fset, filename, content /* src */, parser.AllErrors|parser.ParseComments)
+		progs, err := loadPatches(command.fset, opts, os.Stdin)
+		patchRunner := newPatchRunner(command.fset, progs)
+		_, actualComments, _ := patchRunner.Apply(filename, fast)
+		expectedComments := []string{" This patch replaces instances of fmt.Sprintf()",
+			" with fmt.Errorf()", " Patch files can be applied to mutiple files"}
+		require.NoError(t, err)
+		assert.Equal(t, expectedComments, actualComments)
+	})
+	t.Run("Verify Apply functionality when patch doesn't apply", func(t *testing.T) {
+		_, opts := newArgParser()
+		filename := "testdata/test_files/lint_example/time.go"
+		content, _ := ioutil.ReadFile(filename)
+		fast, _ := parser.ParseFile(command.fset, filename, content /* src */, parser.AllErrors|parser.ParseComments)
+		progs, _ := loadPatches(command.fset, opts, os.Stdin)
+		patchRunner := newPatchRunner(command.fset, progs)
+		_, actualComments, ok := patchRunner.Apply(filename, fast)
+		expectedComments := []string(nil)
+		assert.Equal(t, expectedComments, actualComments)
+		assert.Equal(t, false, ok)
+	})
+
+}
+
+func TestPreviewer(t *testing.T) {
+	t.Parallel()
+	t.Run("run functionality with -l", func(t *testing.T) {
+		var tests = []struct {
+			name           string
+			give           []string
+			want           string
+			targetFilename string
+		}{
+			{
+				name:           "Verify preview's functionality",
+				give:           []string{"single line", "different line"},
+				targetFilename: "testdata/test_files/lint_example/time.go",
+				want: "\n" + "--- testdata/test_files/lint_example/time.go\n" +
+					"+++ testdata/test_files/lint_example/time.go\n@@ -1,1 " +
+					"+0,0 @@\n-single line\n@@ -0,0 +1,1 @@\n+different line\n",
+			},
+			{
+				name:           "Verify preview's functionality - empty file",
+				give:           []string{"", ""},
+				targetFilename: "testdata/test_files/lint_example/time.go",
+				want: "\n" + "--- testdata/test_files/lint_example/time.go\n" +
+					"+++ testdata/test_files/lint_example/time.go\n",
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				buf := bytes.Buffer{}
```

Comment:
style: prefer using `var` style to initialize zero value structs.

```
var buf bytes.Buffer
```

https://github.com/uber-go/guide/blob/master/style.md#use-var-for-zero-value-structs